x/mongo/driver/topology/topology.go

```diff
@@ -297,6 +297,12 @@ func (t *Topology) SelectServer(ctx context.Context, ss description.ServerSelect
 	if atomic.LoadInt32(&t.connectionstate) != connected {
 		return nil, ErrTopologyClosed
 	}
+
+	desc := t.Description()
+	if desc.CompatibilityErr != nil {
```

Comment:
Recall that ServerSelection is not a single check and can block for up to 30 seconds by default. During this time, the topology can be updated and become incompatible if new servers with incompatible wire versions are added. Can we move this check to a place where it's done for every loop iteration?

Reply:
I've moved this check to the beginning of the loop during server selection